vizro-core/examples/_chart-gallery/utils/custom_extensions.py

```diff
@@ -0,0 +1,116 @@
+"""Contains custom components and charts used inside the dashboard."""
+
+from typing import List, Literal
+
+import dash_bootstrap_components as dbc
+import pandas as pd
+import plotly.graph_objects as go
+import vizro.models as vm
+from dash import dcc, html
+from vizro.models.types import capture
+
+try:
+    from pydantic.v1 import Field
+except ImportError:  # pragma: no cov
+    from pydantic import Field
+
+
+class CodeClipboard(vm.VizroBaseModel):
+    """Contains code snippet with a copy to clipboard button."""
+
+    type: Literal["code_clipboard"] = "code_clipboard"
+    title: str = "Code"
+    text: str
+
+    def build(self):
+        """Returns the code clipboard component inside an accordion."""
+        return dbc.Accordion(
+            [
+                dbc.AccordionItem(
+                    dbc.Card(
+                        [
+                            html.H3(self.title),
+                            dcc.Markdown(self.text, id=self.id, className="code-block"),
+                            dcc.Clipboard(target_id=self.id, className="code-clipboard"),
+                        ]
+                    ),
+                    title="SHOW CODE",
+                )
+            ],
+            start_collapsed=True,
+        )
+
+
+class Markdown(vm.VizroBaseModel):
+    """Markdown component."""
+
+    type: Literal["markdown"] = "markdown"
+    text: str = Field(
+        ..., description="Markdown string to create card title/text that should adhere to the CommonMark Spec."
+    )
+    classname: str = ""
+
+    def build(self):
+        """Returns a markdown component with an optional classname."""
+        return dcc.Markdown(id=self.id, children=self.text, dangerously_allow_html=False, className=self.classname)
+
+
+class FlexContainer(vm.Container):
+    """Custom flex `Container`."""
+
+    type: Literal["flex_container"] = "flex_container"
+
+    def build(self):
+        """Returns a flex container."""
+        return html.Div(
+            id=self.id, children=[component.build() for component in self.components], className="flex-container"
+        )
+
+
+@capture("graph")
+def butterfly(data_frame: pd.DataFrame, x1: str, x2: str, y: str) -> go.Figure:
```

Comment:
Take a look here: https://github.com/mckinsey/vizro/pull/559/commits/5d64ec608b2926a011ae38f10055be9e4d907c12

I did not add all of the explanations for the arguments in there, but referenced the Plotly documentation instead. In the end, we do not want to repeat all of the descriptions from the Plotly documentation. I am also wondering what the quality standard in terms of documentation should be e.g. for demos vs. our docs vs. components living in our src code 🤔 So far I've been quite relaxed when it comes to demos, but I definitely get that it might not be intuitive e.g. for the butterfly chart. I didn't intend this to be for complete re-use though, but more like a quick mock-up of a custom chart that works for this demo 😄

Reply:
It's a tricky one - I have not checked how it looks live, but I like the description, and would feel more comfortable using this example.